plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/TestDeltaLakeAnalyze.java

```diff
@@ -327,9 +323,23 @@ public void testAnalyzeSomeColumns()
         assertQueryFails("ANALYZE " + tableName,
                 "List of columns to be analyzed must be a subset of previously used. To extend list of analyzed columns drop table statistics");
 
-        // insert modified rows
+        // insert modified rows should update stats only for already used columns
         assertUpdate("INSERT INTO " + tableName + " SELECT nationkey + 25, concat(name, '1'), regionkey + 5, concat(comment, '21') FROM tpch.sf1.nation", 25);
 
+        assertQuery(
+                "SHOW STATS FOR " + tableName,
+                "VALUES " +
+                        "('nationkey', null, 50.0, 0.0, null, 0, 49)," +
+                        "('regionkey', null, 10.0, 0.0, null, 0, 9)," +
+                        "('comment', null, null, 0.0, null, null, null)," +
+                        "('name', null, null, 0.0, null, null, null)," +
+                        "(null, null, null, null, 50.0, null, null)");
+
+        // insert should not extend list of analyzed columns
+        assertQueryFails(
+                format("ANALYZE %s WITH(columns = ARRAY['nationkey', 'regionkey', 'name'])", tableName),
```

Comment:
pre-existing: 

Running
```
assertUpdate(format("ANALYZE %s WITH(columns = ARRAY['nationkey', 'regionkey'])", tableName));
```
succeeds (as expected).

Now running any of the commands:

```
assertUpdate(format("ANALYZE %s WITH(columns = ARRAY['nationkey'])", tableName));
```
or 
```
assertUpdate(format("ANALYZE %s", tableName));
```

fails with the message:

```
List of columns to be analyzed must be a subset of previously used. To extend list of analyzed columns drop table statistics
```

Now the user needs to know what columns were actually analyzed before to match them exactly in the `ANALYZE` statement.
Maybe we should add in the exception message (separate PR) more information about the columns currently having stats.